ci/scripts/PKGBUILD

```diff
@@ -80,9 +80,13 @@ build() {
     export LIBS="-L${MINGW_PREFIX}/libs"
     export ARROW_S3=OFF
     export ARROW_WITH_RE2=OFF
+    # Without this, some dataset functionality segfaults
+    export CMAKE_UNITY_BUILD=ON
   else
     export ARROW_S3=ON
     export ARROW_WITH_RE2=ON
+    # Without this, some compute functionality segfaults
+    export CMAKE_UNITY_BUILD=OFF
```

Comment:
You mean it segfaults during compilation?

Reply:
It segfaults in the tests. I wasn't really able to debug this on Windows; it disappears once you build with debuginfo.